### ISSUE TITLE
Duplicated issues

### ISSUE THREAD

**Issue by cgalvarez** [phpmd/phpmd#467]
I found duplicated issues, specifically for the rule `CamelCaseParameterName` (don't know if it happens with other rules), due to non-specific `beginline` and `endline` attributes. Maybe it's related to #223, but the errors reported in that issue are different.

Reproducible steps:

1. Create a PHP file `test.php` with these contents:
```php
<?php
function cge_theme_enqueue_styles() {
	$parent_style = 'twentyfifteen-style';
	wp_enqueue_style( $parent_style, get_template_directory_uri() . '/style.css' );
	wp_enqueue_style(
		'cge-site-style',
		get_stylesheet_directory_uri() . '/style.css',
		array( $parent_style ),
		wp_get_theme()->get( 'Version' )
	);
}
```
2. Execute `phpmd test.php xml controversial`. You get the following output:
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<pmd version="@project.version@" timestamp="2017-05-09T11:36:35+00:00">
  <file name="/opt/cgalvarez/wp-theme-cge-site/test.php">
    <violation beginline="2" endline="11" rule="CamelCaseVariableName" ruleset="Controversial Rules" package="+global" externalInfoUrl="#" function="cge_theme_enqueue_styles" priority="1">
      The variable $parent_style is not named in camelCase.
    </violation>
    <violation beginline="2" endline="11" rule="CamelCaseVariableName" ruleset="Controversial Rules" package="+global" externalInfoUrl="#" function="cge_theme_enqueue_styles" priority="1">
      The variable $parent_style is not named in camelCase.
    </violation>
    <violation beginline="2" endline="11" rule="CamelCaseVariableName" ruleset="Controversial Rules" package="+global" externalInfoUrl="#" function="cge_theme_enqueue_styles" priority="1">
      The variable $parent_style is not named in camelCase.
    </violation>
  </file>
</pmd>
```

As you can see, the variable `$parent_style is used` 3 times, at lines 3, 4 and 8, so I think the `beginline` and `endline` attributes should not be set to the function scope where the variable is used, but to the real line in which the variable is used.

My environment:

- PHPMD 2.6.0
- PHP 7.1.4
- OS: Ubuntu Xenial 16.04

**Comment by ravage84** [comment]
Either it should show each line separately with the correct ``beginline`` and ``endline`` set or it should only be displayed once for the whole function.

**Comment by pointlessone** [comment]
It seems the issue is reported on the [wrong node](https://github.com/phpmd/phpmd/blob/bdefdd462ccfa892f9050a712a7ab4ed5f7178b0/src/main/php/PHPMD/Rule/Controversial/CamelCaseVariableName.php#L101). The whole function node instead of just the variable node.
